common/djangoapps/third_party_auth/management/commands/saml.py

```diff
@@ -76,45 +72,146 @@ def _handle_pull_metadata(self):
+        # .. custom_attribute_name: saml_management_command.operation
+        # .. custom_attribute_description: Records current SAML operation ('run_checks').
+        set_custom_attribute('saml_management_command.operation', 'run_checks')
+
+        # .. custom_attribute_name: saml_management_command.site_filter
+        # .. custom_attribute_description: Records the site filter applied, either specific site ID or 'all'.
+        set_custom_attribute('saml_management_command.site_filter', str(site_id) if site_id else 'all')
+
+        metrics = self._check_provider_configurations(site_id)
+        self._report_check_summary(metrics)
+
+    def _check_provider_configurations(self, site_id):
+        """
+        Check each provider configuration for potential issues.
+        Returns a dictionary of metrics about the found issues.
+        """
+        outdated_count = 0
+        site_mismatch_count = 0
+        slug_mismatch_count = 0
+        null_config_count = 0
         error_count = 0
+        total_providers = 0
 
-        # Filter by site if specified
         provider_configs = SAMLProviderConfig.objects.current_set()
         if site_id:
             provider_configs = provider_configs.filter(site_id=site_id)
 
+        self.stdout.write(self.style.SUCCESS("SAML Configuration Check Report"))
+        self.stdout.write("=" * 50)
+
         for provider_config in provider_configs:
-            if provider_config.saml_configuration:
-                try:
-                    current_config = SAMLConfiguration.current(
-                        provider_config.site_id,
-                        provider_config.saml_configuration.slug
-                    )
+            total_providers += 1
+            provider_info = (
+                f"Provider '{provider_config.slug}' "
+                f"(ID: {provider_config.id}, site {provider_config.site_id})"
+            )
+
+            if not provider_config.saml_configuration:
+                self.stdout.write(f"[INFO] {provider_info} has no SAML configuration (may be intentional)")
+                null_config_count += 1
+                continue
 
-                    if current_config and current_config.id != provider_config.saml_configuration_id:
+            try:
+                current_config = SAMLConfiguration.current(
+                    provider_config.saml_configuration.site_id,
+                    provider_config.saml_configuration.slug
+                )
+
+                # Check for outdated configuration references
+                if current_config:
+                    if current_config.id != provider_config.saml_configuration_id:
                         self.stdout.write(
-                            f"Provider '{provider_config.slug}' (site {provider_config.site_id}) "
+                            f"[OUTDATED] {provider_info} "
                             f"has outdated config (ID: {provider_config.saml_configuration_id} -> {current_config.id})"
```

Comment:
1. I'd rather just use the more traditional "[WARNING]". 
2. If you try to update the SAML config using the UI, is the id enough, or would you need other details?
3. I made it more wordy, but people would need to manually update.
```suggestion
                            f"[WARNING] {provider_info} "
                            f"has outdated SAML config (id={provider_config.saml_configuration_id} which "
                            f"should be updated to the current SAML config (id={current_config.id})."
```